apps/sertra/components/mod-selector/search.tsx

```diff
@@ -16,18 +16,19 @@ export const SearchBox = <T,>({
   keyExtractor,
   options,
   onSelect,
+  searchVars,
 }: SearchBoxProps<T>) => {
   const [isFocused, setFocused] = useState<boolean>(false);
 
   const [filteredOptions, setfilteredOptions] = useState(options);
   const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
     const results = options.filter((option) => {
-      if (option.ownerName.toLowerCase().includes(e.target.value.toLowerCase()))
-        return option;
-      if (
-        option.packageName.toLowerCase().includes(e.target.value.toLowerCase())
-      )
-        return option;
+      for (const searchVar of searchVars) {
+        if (
+          option[searchVar].toLowerCase().includes(e.target.value.toLowerCase())
```

Comment:
My first comment gives us a better change that `option[searchVar]` is actually a thing. But it might be e.g. an int, bool, or undefined, e.g. if the interface declares the value an optional string. Before we can call `.toLowerCase`, which only exists for strings, we need to check that `option[searchVar]` is a string:

```
        const field = option[searchVar];
        if (
          typeof field === "string" &&
          field.toLowerCase().includes(searchValue)
        ) {
          return option;
        }
```

And please use braces around the block even if it's just a single line. That's something that should be caught by linter, but you know the current situation.

Reply:
As a side note, this search implementation doesn't work if user tries to search with something like "teamname modname". But I think this is good enough for the MVP.